박민영/Graph Search (그래프 탐색)/DFS (깊이 우선 탐색)/백준/2468.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * @author Minyeong Park
+ * @date 2023.03.16.
+ * 안전 영역
+ * 문제 링크 : https://www.acmicpc.net/problem/2468
+ */
+
+
+
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class Main {
+    static int n;
+    static int[][] arr;
+    static boolean[][] visited;
+    static int[] dx = {1, 0, -1, 0};
+    static int[] dy = {0, -1, 0, 1};
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        n = Integer.parseInt(br.readLine());
+        arr = new int[n][n]; // (n+1)*(n+1)로 하려고 했었는데 그냥 n*n으로 설정하면 됨
+        StringTokenizer st;
+        int max_height = 0;
+        for (int i = 0; i < n; i++) {
+            st = new StringTokenizer(br.readLine());
+            for (int j = 0; j < n; j++) {
+                arr[i][j] = Integer.parseInt(st.nextToken());
+                max_height = Math.max(max_height, arr[i][j]);
+            }
+        }
+
+        int max = 0;
+        // 1. 모든 지역 탐색
+        for (int h = 0; h < max_height + 1; h++) {
+            visited = new boolean[n][n];
+            int cnt = 0;
+            for (int i = 0; i < n; i++) {
+                for (int j = 0; j < n; j++) {
+                    // 2. 안전 영역 탐지
+                    if (!visited[i][j] && arr[i][j] > h) {
+                        cnt += dfs(i, j, h); // 해당 안전 영역 탐색 시작
+                    }
+                }
+            }
+            max = Math.max(max, cnt);
+        }
+        System.out.println(max);
+    }
+
+    // 안전 영역 DFS 탐색
+    static int dfs(int x, int y, int h) {
+        visited[x][y] = true;
+        for (int i = 0; i < 4; i++) {
+            int nx = x + dx[i];
+            int ny = y + dy[i];
+
+            if (nx < 0 || ny < 0 || nx > n - 1 || ny > n - 1)
+                continue;
+            if (visited[nx][ny])
+                continue;
+            if (arr[nx][ny] > h) {
+                dfs(nx, ny, h);
+            }
```

Comment:
```suggestion
            if (nx < 0 || ny < 0 || nx > n - 1 || ny > n - 1) continue;
            if (visited[nx][ny]) continue;
            if (arr[nx][ny] > h) dfs(nx, ny, h);
```
중요한 건 아니지만 코드 포맷을 통일 시키면 가독성이 좋아질 것 같아요!
저도 코드량을 줄이는 방향으로 코드를 짜려고 노력하고 있어요 🙃

Reply:
오옷 감사합니다! 포맷을 통일시킬 생각은 못했는데 덕분에 배워가요!!:smile: